### PR TITLE
Added search tree component to [WIP] UI

### PR DESCRIPTION
**Pull request by sahirgomez1**
### Added search tree component to [WIP] frontend.

https://user-images.githubusercontent.com/84393628/125869263-d400473c-7a45-4a6d-8b7e-3d647fdd4402.mov

### CODE REVIEW

**Comment by hughleat** [comment]
That looks fantastic! Well done!

File: www/frontends/compiler_gym/src/assets/scss/custom.scss [hunk 1]
```diff
@@ -1,5 +1,6 @@
 @import "./variables.scss";
```
Comment (ChrisCummins) [inline www/frontends/compiler_gym/src/assets/scss/custom.scss:1]: Every source files needs the Facebook copyright / MIT license header. Some of the scss and js files don't have them, could you please add them (I realize I forgot to do that for demo_api.py!)

File: www/frontends/compiler_gym/src/assets/scss/highlightjs.scss [hunk 2]
```diff
@@ -1,6 +1,5 @@
 .ir-state-container {
   margin: 1rem 0;
-  //padding-right: 1rem;
```
Comment (ChrisCummins) [inline www/frontends/compiler_gym/src/assets/scss/highlightjs.scss:3]: 👍

File: www/frontends/compiler_gym/src/components/ActionsNavbar.js [hunk 3]
```diff
@@ -41,9 +41,10 @@ const CustomMenu = forwardRef(
   }
 );
 
+//name.toLowerCase().indexOf(this.props.filter.toLowerCase()) !== -1
+
```
Comment (ChrisCummins) [inline www/frontends/compiler_gym/src/components/ActionsNavbar.js]:
```suggestion
```

(I'm a little pedantic about committing commented-out code hehe)

**Review by ChrisCummins — COMMENTED** [review]
This looks great, thank you @sahirgomez1! I followed the install instructions in the README and ran this locally. I know it's still an early-stage prototype but its super inspiring to think of all the possibilities that this GUI will unlock!

Some nitpicky comments (not all of which need fixing for this PR):

- In the "controls" section, rename "Benchmark" to "Dataset", and "Uri" to "Benchmark".
- Can we set it so that when the user changes Dataset, the Benchmark immediately switches to some default, such as the first element in the list?
- When I close the tab, chrome pops up a message "Leave Site? Changes you made may not be saved". Is there a way to disable that?
- Consider moving `www/frontends/README.md` to `www/README.md`.
- In the Installation instructions of the README, could you add the necessary install step for linux/macos to install npm?

Regarding the 3 failing GitHub checks:

- [CI/bazel_test](https://github.com/facebookresearch/CompilerGym/pull/319/checks?check_run_id=3081662028): failures are caused by some irrelevant breakage, we can ignore them. If you like you can pull and rebase on the latest `www` (I merged a fix in #315).
- [Pre-commit](https://github.com/facebookresearch/CompilerGym/runs/3081662013?check_suite_focus=true): These are code style checkers that will enforce things like no trailing whitespace, python PEP8 format, etc. I would recommend setting your local development environment to run these checkers automatically as a pre-commit git hook, see [here](https://github.com/facebookresearch/CompilerGym/blob/development/.pre-commit-config.yaml#L4-L30) for details. In most cases, the checkers will fix the problems for you.

Cheers,
Chris

File: www/frontends/compiler_gym/src/assets/scss/custom.scss (hunk 1 above)
Comment (sahirgomez1) [reply www/frontends/compiler_gym/src/assets/scss/custom.scss:1]: Noted!

File: www/frontends/compiler_gym/src/assets/scss/highlightjs.scss (hunk 2 above)
Comment (sahirgomez1) [reply www/frontends/compiler_gym/src/assets/scss/highlightjs.scss:3]: Noted

File: www/frontends/compiler_gym/src/components/ActionsNavbar.js (hunk 3 above)
Comment (sahirgomez1) [reply www/frontends/compiler_gym/src/components/ActionsNavbar.js]: Noted and fixed

**Review by ChrisCummins — APPROVED** [review]
Thanks for quickly resolving those comments @sahirgomez1. LGTM!

**Pull request merged**